src/wp-includes/class-wp-scripts.php

```diff
@@ -281,6 +281,29 @@ public function do_item( $handle, $group = false ) {
 			$intended_strategy = '';
 		}
 
+		/*
+		 * Move this script to the footer if:
+		 * 1. The script is in the header group.
+		 * 2. The current output is the header.
+		 * 3. The intended strategy is delayed.
+		 * 4. The actual strategy is not delayed.
+		 * 5. All dependent scripts are in the footer.
+		 */
+		if ( $intended_strategy && ! $this->is_delayed_strategy( $strategy ) && 0 === $this->groups[ $handle ] && 0 === $group ) {
+			// Check that all depedencies are in the footer.
+			$all_deps_in_footer = true;
+			foreach ( $this->get_dependents( $handle ) as $dep ) {
+				if ( isset( $this->groups[ $dep ] ) && $this->groups[ $dep ] > 0 ) {
+					$all_deps_in_footer = false;
+					break;
+				}
+			}
+			if ( $all_deps_in_footer ) {
+				$this->in_footer[] = $handle;
```

Comment:
Does this no longer need to also do `$this->set_group( $handle, 1 );`? It's not needed above when something was registered already as being in the footer, but if we're _moving_ something to the footer, it would seem needed. I guess not because of this condition in `set_group`:

https://github.com/WordPress/wordpress-develop/blob/3ad5b1b9360570becb6272b1730d1e6f1e44893f/src/wp-includes/class-wp-scripts.php#L646-L648

So if something didn't get printed in the `head`, it will end up still getting printed in the footer because at the footer group is `1` which is greater than the head group of `0`.

Reply:
Hmm, right - ok that makes sense I will restore this. Also could be useful if a handle were checked later in a dependency check.

Reply:
Actually, I thought it made sense that it _wasn't_ needed.

Reply:
Ah I see. I still think it is worth having here for two possible cases: 
- a subsequent dependency check looks at group to see if this script is in the footer (it is since we moved it there)
- a plugin checks groups for some reason after script output

Reply:
Ok, makes sense 👍

Reply:
Oh, but you just re-committed the removal?

Reply:
OK, after adding assertions in 7ad534425808c0a7772914cdc2fe0088398ddb5c and then trying to re-add the call to `set_group`, I also see that it does nothing. It seems this code in `WP_Scripts::set_group()` is responsible for it not being set:

https://github.com/WordPress/wordpress-develop/blob/93782b51775d79d0547c1d551fc236dd4274bbc3/src/wp-includes/class-wp-scripts.php#L623-L625

And in `WP_Dependencies::set_group()`:

https://github.com/WordPress/wordpress-develop/blob/93782b51775d79d0547c1d551fc236dd4274bbc3/src/wp-includes/class-wp-dependencies.php#L485-L487

If I try to force it to be set, then it causes everything to get moved to the `head` though,  so that's clearly wrong. Seems like we should go ahead and keep it omitted. I will note that it seems the proper call is:

```php
$this->set_group( $handle, false, 1 )
```

Where the second argument is `$recursion` and the third argument is the actual group. I see in 039d76a0d8b3a13db77b0dfe53b64da9dfa36b4f that you removed:

```php
$this->set_group( $handle, 1 );
```

But even with the proper args, it doesn't do what is expected.

Reply:
So yeah, I think we should just let's leave it out! 😆